SteakBot.Core/Modules/HelpModule.cs

```diff
@@ -24,9 +27,56 @@ public HelpModule(MemeService memeService, CommandService commandService)
         [Command("list")]
         [Summary("Lists all available \"memes\"")]
         public async Task List()
+        {
+            await ReplyMemeCommands(_memeService.MemeCommands);
+        }
+
+        [Command("list")]
+        [Summary("Filters and lists available \"memes\"")]
+        [Remarks("Usage: `list <filter(text)>`")]
+        public async Task List(string filter)
+        {
+            var memeCommands = _memeService.MemeCommands;
+            if (!string.IsNullOrWhiteSpace(filter))
+            {
+                memeCommands = memeCommands
+                    .Where(x => x.Name.Contains(filter) ||
+                                x.Description.Contains(filter))
+                    .ToList();
+            }
+
+            await ReplyMemeCommands(memeCommands);
+        }
+
+        [Command("help")]
+        [Summary("Lists all commands with their summaries")]
+        public async Task Help()
+        {
+            await ReplyHelpMessage(_commandService.Commands);
+        }
+
+        [Command("help")]
+        [Summary("Filters and lists commands with their summaries")]
+        [Remarks("Usage: `help <filter(text)>`")]
+        public async Task Help(string filter)
+        {
+            var commands = _commandService.Commands;
+            if (!string.IsNullOrWhiteSpace(filter))
+            {
+                commands = commands
+                    .Where(x => x.Name.Contains(filter) ||
+                                (x.Remarks != null && x.Remarks.Contains(filter)));
```

Comment:
This should also check the commands' Summary.

Reply:
Done.